<!-- Phidgets IMU launch file -->
<launch>

  <!-- Nodelet manager -->
  <node pkg="nodelet" type="nodelet" name="imu_manager" 
    args="manager" output="screen" respawn="true"/>

  <!-- IMU Driver -->
  <node pkg="nodelet" type="nodelet" name="PhidgetsImuNodelet" args="load phidgets_imu/PhidgetsImuNodelet imu_manager" output="screen" respawn="true">

	<param name="frame_id" value="base_link"/>

    <!-- supported data rates: 4 8 16 24 32 40 ... 1000 (in ms) -->
    <param name="period" value="4"/>

    <!-- compass calibration parameters -->
	<param name="cc_mag_field" value="0.507681"/>
    <param name="cc_offset0" value="-0.074"/>
	<param name="cc_offset1" value="0.177"/>
	<param name="cc_offset2" value="-0.052"/>
	<param name="cc_gain0" value="2.303"/>
	<param name="cc_gain1" value="2.291"/>
	<param name="cc_gain2" value="2.648"/>
	<param name="cc_t0" value="0.415195"/>
	<param name="cc_t1" value="-0.077192"/>
	<param name="cc_t2" value="0.175301"/>
	<param name="cc_t3" value="-0.052127"/>
	<param name="cc_t4" value="2.259510"/>
	<param name="cc_t5" value="2.307828"/>
	<param name="cc_t6" value="2.658185"/>
	<param name="cc_t7" value="-0.008807"/>
	<param name="cc_t8" value="-0.009136"/>
	<param name="cc_t9" value="-0.008999"/>
	<param name="cc_t10" value="-0.011983"/>
	<param name="cc_t11" value="-0.010375"/>
	<param name="cc_t12" value="-0.013803"/>

  </node>

  <!-- IMU Orientation Filter (Madgwick) -->
  <node pkg="nodelet" type="nodelet" name="ImuFilterNodelet" args="load imu_filter_madgwick/ImuFilterNodelet imu_manager" output="screen" respawn="true">

	<param name="world_frame" value="nwu"/>
	<param name="fixed_frame" value="base_stabilized"/>
    <param name="use_mag" value="true"/>
    <param name="use_magnetic_field_msg" value="true"/>

  </node>
  <!-- LIDAR Launch -->
  <node name="urg_node" pkg="urg_node" type="urg_node" output="screen" respawn="true">
    <param name="ip_address" value="192.168.0.10"/>
    <param name="serial_port" value="/dev/ttyACM0"/>
    <param name="serial_baud" value="115200"/>
    <param name="frame_id" value="laser"/>
    <param name="calibrate_time" value="true"/>
    <param name="publish_intensity" value="false"/>
    <param name="publish_multiecho" value="false"/>
    <param name="angle_min" value="-1.6580628"/>
    <param name="angle_max" value="1.6580628"/>
</node>

	<!-- visualization launch-->
	<arg name="model" default="$(find div_datalogger_postprocess)/urdf/DataLogger.urdf" />
	<arg name="gui" default="true" />
	<arg name="rvizconfig" default="$(find div_datalogger_postprocess)/rviz/postprocessing.rviz" />
	<param name="use_sim_time" value="false"/>

	<!-- URDF 3D model -->
  	<param name="robot_description" command="$(find xacro)/xacro.py $(arg model)" />
  	<param name="use_gui" value="$(arg gui)"/>

  	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  	<node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
  	
  	<!-- Point cloud assembly -->
  	<!-- for package laseÅ• assembler -->
  	
  	<!--<node type="laser_scan_assembler" pkg="laser_assembler" name="my_assembler">
    	<remap from="scan" to="/scan"/>
    	<param name="max_scans" type="int" value="100" />
    	<param name="fixed_frame" type="string" value="base_stabilized" />
    	<param name="ignore_laser_skew" type="bool" value="false" />
  	</node>
  	<node type="periodic_snapshotter" name="periodic_snapshotter" pkg="laser_assembler"></node>-->
 
  	<!-- RVIZ -->
  	<!--<node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />-->
</launch>
